signer/tests/integration/transaction_coordinator.rs

```diff
@@ -1115,6 +1117,242 @@ async fn run_dkg_from_scratch() {
+            ev.run().await
+        });
+    });
+
+    tx_signer_processes.for_each(|ev| {
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            ev.run().await
+        });
+    });
+
+    while start_count.load(Ordering::SeqCst) < 6 {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+
+    // 5. Once they are all running, signal that DKG should be run. We
+    //    signal them all because we do not know which one is the
+    //    coordinator.
+    signers.iter().for_each(|(ctx, _, _, _)| {
+        ctx.get_signal_sender()
+            .send(RequestDeciderEvent::NewRequestsHandled.into())
+            .unwrap();
+    });
+
+    // Await the `stacks_tx_receiver_task` to receive the first transaction broadcasted.
+    let broadcast_stacks_txs =
+        tokio::time::timeout(Duration::from_secs(10), stacks_tx_receiver_task)
+            .await
+            .unwrap()
+            .expect("failed to receive message")
+            .expect("no message received");
+
+    // A BTreeSet to uniquely hold all the aggregate keys we find in the database.
+    let mut all_aggregate_keys = BTreeSet::new();
+
+    for (_, db, _, _) in signers.iter() {
+        // Get the aggregate keys from this signer's database.
+        let mut aggregate_keys =
+            sqlx::query_as::<_, (PublicKey,)>("SELECT aggregate_key FROM sbtc_signer.dkg_shares")
+                .fetch_all(db.pool())
+                .await
+                .unwrap();
+
+        // 6. Check that we have exactly one row in the `dkg_shares` table.
+        assert_eq!(aggregate_keys.len(), 2);
+        for key in aggregate_keys.iter() {
+            all_aggregate_keys.insert(key.0.clone());
+        }
+
+        // An additional sanity check that the query in
+        // get_last_encrypted_dkg_shares gets the right thing (which is the
+        // only thing in this case).
+        let key = aggregate_keys.pop().unwrap().0;
+        let shares = db.get_latest_encrypted_dkg_shares().await.unwrap().unwrap();
+        assert_eq!(shares.aggregate_key, key);
+    }
+
+    // 7. Check that they all have the same aggregate keys in the
+    //    `dkg_shares` table.
+    assert_eq!(all_aggregate_keys.len(), 2);
+
+    // 8. Check that the coordinator broadcast a rotate key tx
+    broadcast_stacks_txs.verify().unwrap();
+
+    let TransactionPayload::ContractCall(contract_call) = broadcast_stacks_txs.payload else {
+        panic!("unexpected tx payload")
+    };
+    assert_eq!(
+        contract_call.contract_name.to_string(),
+        RotateKeysV1::CONTRACT_NAME
+    );
+    assert_eq!(
+        contract_call.function_name.to_string(),
+        RotateKeysV1::FUNCTION_NAME
+    );
+    let rotate_keys = RotateKeysV1::new(
+        &signer_wallet,
+        signers.first().unwrap().0.config().signer.deployer,
+        all_aggregate_keys.iter().next().unwrap(),
```

Comment:
I think this may be fragile, right? This has both aggregate keys, and there shouldn't be any guarantee that the first one, when sorted, is the most recently generated one. Better to just check that it is the latest aggregate key.

Reply:
Well, it wasn't, but I did get it to fail by setting the agg key explicitly to `aggregate_key_1` -- not sure why it doesn't fail like this 🤔 Anyway, changed it to explicitly use the new key in [680d822](https://github.com/stacks-network/sbtc/pull/1212/commits/680d82203c7c447d51ab8f84aa8d0b1685c533b5)

Reply:
I think that it was the random number generator that made it less fragile than it looks. The aggregate key that is constructed using the pseudo-randomness from `rand::rngs::StdRng::seed_from_u64(51)` must have been quite large. Switching to the `OsRng` gives failures at the expected frequency.